include/boost/url/detail/any_segments_iter.hpp

```diff
@@ -40,6 +40,12 @@ struct BOOST_SYMBOL_VISIBLE
     // to handle special cases
     string_view front;
 
+    // quick number of segments
+    // 0 = zero
+    // 1 = one
+    // 2 = two, or more
+    std::size_t fast_nseg = 0;
```

Comment:
Why do we need `std::size_t` for 3 values?

Reply:
you can make it smaller if you want, but it won't change the size of the struct. an int would be good